safe/impact_function_v4/impact_function.py

```diff
@@ -644,7 +652,10 @@ def prepare(self):
                 )
                 return PREPARE_FAILED_BAD_INPUT, message
 
+            self._provenance['exposure_layer'] = self.exposure.source()
```

Comment:
Really? I got the correct one actually.
I put here, in the prepare method to avoid getting the not-original source one when we call run (since there are some layer assignments inside the method).